### DISCUSSION TITLE
Integrationsproblem: cannot import name 'SensorDeviceClass' from ...

### DISCUSSION THREAD

**Discussion by wora55** [weltenwort/home-assistant-rct-power-integration#197] [Q&A]
ich habe custom_components angelegt und files hineinkopiert und System neu gestartet:
RCT-Client war nicht anwählbar über Oberfläche.
Dann habe ich das Script check_config gestartet und es hat mich mit folgenden Fehler verblüfft:

```
INFO:homeassistant.util.package:Attempting install of rctclient==0.0.3
Failed config
  General Errors:
    - Component error: rct_power - cannot import name 'SensorDeviceClass' from 'homeassistant.components.sensor' (/srv/homeassistant/lib/python3.9/site-packages/homeassistant/components/sensor/__init__.py)

```
Der Identifier "SensorDeviceClass" ist wirklich nicht in dem File, das habe ich kontrolliert.
Allerdings auch in keinem anderen .py-File unter components/  (sagt grep SensorDeviceClass $(find . -name '*.py'))

Referenziert wird der Typ hier:

`pi@cassiopeia-pi4:/home/homeassistant/.homeassistant $ grep SensorDeviceClass $(find custom_components -name '*.py')
custom_components/rct_power/lib/device_class_helpers.py:from homeassistant.components.sensor import SensorDeviceClass
custom_components/rct_power/lib/device_class_helpers.py:        return SensorDeviceClass.TEMPERATURE
custom_components/rct_power/lib/device_class_helpers.py:        return SensorDeviceClass.VOLTAGE
custom_components/rct_power/lib/device_class_helpers.py:        return SensorDeviceClass.CURRENT
custom_components/rct_power/lib/device_class_helpers.py:        return SensorDeviceClass.POWER
custom_components/rct_power/lib/device_class_helpers.py:        return SensorDeviceClass.ENERGY
custom_components/rct_power/lib/entities.py:from homeassistant.components.sensor import SensorDeviceClass, SensorStateClass
custom_components/rct_power/lib/entities.py:        device_class=SensorDeviceClass.BATTERY,
`
Was kann/muss ich tun, um das zum Laufen zu bringen?

Gefunden: Update von core auf core22 durchgeführt

**Comment by weltenwort** [comment]
Hallo @wora55, ich weiß nicht was dort schief läuft. Das importierte enum `SensorDeviceClass` existiert im source code:

https://github.com/home-assistant/core/blob/2022.3.5/homeassistant/components/sensor/__init__.py#L73

> Gefunden: Update von core auf core22 durchgeführt

Was genau meinst du damit? Was ist `core22`?

**Comment by wora55** [comment]
Problem geklärt.
Ich hatte eine alte Version des HA-core (21?).
Seit ich die aktualisiert habe, funktioniert auch alles.
Herausgefunden habe ich das, weil in Deinen Anmerkungen irgendwo steht, dass Du im Ende 2021 irgend etwas aktualisieren musstest, weil sich in  HA etwas geändert hat.
Bis zu dem Zeitpunkt dachte ich immer HA aktualisiert sich selbst ...


Trotzdem danke. Ich bin von dem Paket fast komplett begeistert. Habe/hatte nur in einem Detail ein kleines Problem. Beschreibe ich genauer und schicke es Dir dann. Ist kein "Fehler" in Deinem Teil sondern einfach nur schwierig in HA einzubinden.


Wolfgang
 

Felix Stürmer schrieb am Donnerstag, 17. März 2022 17:54:47 (+01:00):



Hallo @wora55, ich weiß nicht was dort schief läuft. Das importierte enum SensorDeviceClass existiert im source code:

https://github.com/home-assistant/core/blob/2022.3.5/homeassistant/components/sensor/__init__.py#L73

Gefunden: Update von core auf core22 durchgeführt

Was genau meinst du damit? Was ist core22?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
You are receiving this because you were mentioned.


-- 
Wolfgang Radke
Schubertstr. 6f
76344 Eggenstein

**Comment by weltenwort** [comment]
Ah, danke für die Info. HA wird nur im "supervised" setup aktualisiert und auch dann nicht automatisch, sondern über das supervisor UI (vgl. https://www.home-assistant.io/installation/#compare-installation-methods). Zum Installieren und Aktualisieren dieser (und anderer) Integrationen würde ich die [HACS](https://hacs.xyz/) Integration empfehlen.

**Comment by wora55** [comment]
Hallo,

habe gerade auf HA 2024.1 upgedatet und HA meldet jetzt, dass die Werte 'unavailable' sind.
Habe System anschließend (komplett) neu gestartet, gleiches Ergebnis
HACS ist auch auf dem aktuellen Stand.

Wo kann ich sinnvoll nach möglichen Fehlermeldungen schauen?

Wolfgang
